src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -45,11 +45,13 @@ public JsonSerializerOptions()
         /// <summary>
         /// Defines whether an extra comma at the end of a list of JSON values in an object or array
         /// is allowed (and ignored) within the JSON payload being deserialized.
-        /// By default, it's set to false, and <exception cref="JsonException"/> is thrown if a trailing comma is encountered.
         /// </summary>
         /// <exception cref="InvalidOperationException">
         /// Thrown if this property is set after serialization or deserialization has occurred.
         /// </exception>
+        /// <remarks>
+        /// By default, it's set to false, and <exception cref="JsonException"/> is thrown if a trailing comma is encountered.
+        /// </remarks>
         public bool AllowTrailingCommas
         {
             get
```

Comment:
This isn't readonly?  I'm not clear on when we're choosing to use it in these types and when we're not.

Reply:
I am not familiar enough with all the `JsonSerializerOptions` (and the new `readonly` feature) to know what getters should have it. Presumably most value type property getters should be marked as readonly.

AllowTrailingCommas
DefaultBufferSize
IgnoreNullValues
IgnoreReadOnlyProperties
MaxDepth
PropertyNameCaseInsensitive
ReadCommentHandling
WriteIndented

I don't know about these two:
```C#
public JsonNamingPolicy PropertyNamingPolicy
public JsonNamingPolicy DictionaryKeyPolicy
```
https://github.com/dotnet/corefx/blob/d1afe6bd921bf33115f52d6eb61753280a42179a/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L145
https://github.com/dotnet/corefx/blob/d1afe6bd921bf33115f52d6eb61753280a42179a/src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs#L101

cc @steveharter

Would property getters on JsonWriterOptions also be eligible?

https://github.com/dotnet/corefx/blob/d1afe6bd921bf33115f52d6eb61753280a42179a/src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterOptions.cs#L26

I am going to address this feedback outside this PR.

Reply:
> Nullability of reference types in type of parameter 'nonPublic' doesn't match overridden member.

Ok

Reply:
Basically, any method which doesn't mutate instance state can be `readonly`.

Members which will never mutate state should be readonly. Property getters generally fall into this realm.

Members which may mutate state but only as part of deterministic lazy initialization (so one time mutation, two threads initiating concurrently will get the same result/etc) can be readonly.

Members which mutate state cannot be readonly.

Reply:
Right, I understand when `readonly` can be used; I was saying I didn't understand the lack of consistency with which we were applying it to these types.

Reply:
> Right, I understand when readonly can be used

I was giving an explanation for @ahsonkhan, since he indicated he didn't understand where they can/should be used :)

Reply:
Ah, ok :)